lib/next_ls/signature_help.ex

```diff
@@ -0,0 +1,117 @@
+defmodule NextLS.SignatureHelp do
+  @moduledoc false
+
+  import NextLS.DB.Query
+
+  alias GenLSP.Structures.ParameterInformation
+  alias GenLSP.Structures.SignatureHelp
+  alias GenLSP.Structures.SignatureInformation
+  alias NextLS.ASTHelpers
+  alias NextLS.DB
+
+  def fetch(file, {line, col}, db) do
+    symbol = fetch_symbol(file, line, col, db)
+
+    case symbol do
+      nil ->
+        nil
+
+      [] ->
+        nil
+
+      [[_, _mod, file, type, label, line, col | _] | _] = _definition ->
+        if type in ["def", "defp"] do
+          code = File.read!(file)
+
+          params =
+            code
+            |> ASTHelpers.Functions.get_function_params(label, line, col)
+            |> Enum.map(fn {name, _, _} ->
+              %ParameterInformation{
+                label: Atom.to_string(name)
+              }
+            end)
```

Comment:
I think an easier route here, to avoid having to parse the source code and find the function, is to embed the parameters into the database. I believe this information should be available in the compiler tracer.

Reply:
That can be done is a separate PR as well as its an orthogonal chane

Reply:
https://github.com/elixir-tools/next-ls/blob/main/lib/next_ls/db.ex#L85